elvish-mode.el

```diff
@@ -61,9 +61,12 @@ determined by the syntax table. This allows us to keep things like '-'
 in the symbol part of the syntax table, so `forward-word' works as
 expected.")
 
+(defconst elvish-start-of-statement '(sequence (or line-start "(" ";" "|") (zero-or-more space))
+  "Regex to match the beginning of an Elvish statement (where a command or keyword can appear)")
+
 (defconst elvish-keyword-pattern
   (let ((keywords (cons 'or elvish-keywords)))
-    (eval `(rx symbol-start (group ,keywords) symbol-end)))
+    (eval `(rx ,elvish-start-of-statement (group ,keywords) symbol-end)))
```

Comment:
My main motivation was to allow proper highlighting of the arithmetic and comparison functions. And you are right that in principle a keyword can appear anywhere - *but* a function can only appear at the start of a statement. How about I remove the `elvish-start-of-statement` from the keyword highlighting pattern, and leave it only for functions?